cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/ZooKeeperReconciler.java

```diff
@@ -207,7 +207,7 @@ public Future<Void> reconcile(KafkaStatus kafkaStatus, Clock clock)    {
                 .compose(i -> serviceEndpointsReady())
                 .compose(i -> headlessServiceEndpointsReady())
                 .compose(i -> deletePersistentClaims())
-                .compose(i -> maybeDeleteControllerZnode());
+                .compose(i -> this.isKRaftMigrationRollback ? maybeDeleteControllerZnode() : Future.succeededFuture());
```

Comment:
if we rename the current `maybeDeleteControllerZnode` to `deleteControllerZnode` (see next comment), I think we should have a new `maybeDeleteControllerZnode` method containing this code, so something like:

```java
protected Future<Void> maybeDeleteControllerZnode() {
    return this.isKRaftMigrationRollback ? maybeDeleteControllerZnode() : Future.succeededFuture();
}
```

so the compose in the reconcile will stay:

```java
.compose(i -> maybeDeleteControllerZnode());
```

This approach makes more understandable the mocking in the test (see related comment later).